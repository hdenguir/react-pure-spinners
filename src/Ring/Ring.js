import React from 'react'
import styles from './styles.css'
const Ring = () => (
  <>
    <style>{`${styles}`}</style>
    <div className="lds-dual-ring"></div>
  </>
)

export default Ring
