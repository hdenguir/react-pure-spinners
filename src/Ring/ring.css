:root {
  --rps-ring-color: #fff;
  --rps-ring-size: 64px;
  --rps-ring-thickness: 6px;
}

.rps-ring {
  display: inline-block;
  position: relative;
  width: var(--rps-ring-size);
  height: var(--rps-ring-size);
}

.rps-ring div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  border: var(--rps-ring-thickness) solid var(--rps-ring-color);
  border-radius: 50%;
  animation: rps-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border-color: var(--rps-ring-color) transparent transparent transparent;
}

.rps-ring div:nth-child(1) {
  animation-delay: -0.45s;
}

.rps-ring div:nth-child(2) {
  animation-delay: -0.3s;
}

.rps-ring div:nth-child(3) {
  animation-delay: -0.15s;
}

@keyframes rps-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
