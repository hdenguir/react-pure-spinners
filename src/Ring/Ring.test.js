import Ring from './Ring'

import React from 'react'

import renderer from 'react-test-renderer'

it('Redners Correctly', () => {
  const tree = renderer.create(<Ring />).toJSON()
  expect(tree).toMatchSnapshot()
})
