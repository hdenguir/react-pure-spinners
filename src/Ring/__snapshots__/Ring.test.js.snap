// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redners Correctly 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="lds-dual-ring"
  />,
]
`;
